src/core/electric_utility.jl

```diff
@@ -67,27 +88,101 @@ function ElectricUtility(;
+        emissions_factor_SO2_decrease_pct::Float64 = 0.01174,
+        emissions_factor_PM25_decrease_pct::Float64 = 0.01174,
         outage_start_time_step::Int=0,  # for modeling a single outage, with critical load spliced into the baseline load ...
         outage_end_time_step::Int=0,  # ... utiltity production_factor = 0 during the outage
-        allow_simultaneous_export_import::Bool = true,  # if true the site has two meters (in effect)
-        # variables below used for minimax the expected outage cost,
+        allow_simultaneous_export_import::Bool=true,  # if true the site has two meters (in effect)
+        # next 5 variables below used for minimax the expected outage cost,
         # with max taken over outage start time, expectation taken over outage duration
         outage_start_time_steps::Array{Int,1}=Int[],  # we minimize the maximum outage cost over outage start times
         outage_durations::Array{Int,1}=Int[],  # one-to-one with outage_probabilities, outage_durations can be a random variable
-        outage_probabilities::Array{R,1} where R<:Real = [1.0],
+        outage_probabilities::Array{<:Real,1}=[1.0],
         outage_time_steps::Union{Missing, UnitRange} = isempty(outage_durations) ? missing : 1:maximum(outage_durations),
         scenarios::Union{Missing, UnitRange} = isempty(outage_durations) ? missing : 1:length(outage_durations),
         net_metering_limit_kw::Real = 0,
-        interconnection_limit_kw::Real = 1.0e9
-    )
+        interconnection_limit_kw::Real = 1.0e9,
+        latitude::Real,
+        longitude::Real,
+        time_steps_per_hour::Int = 1
+        )
+
+        region_abbr, meters_to_region = region_abbreviation(latitude, longitude)
+        emissions_region = region(region_abbr)
+        emissions_series_dict = Dict{String,Array{Float64}}()
+
+        #TODO: can this section be refactored by emissions_type by using Symbol("") technique?
+        #eval(Meta.parse("emissions_factor_series_lb_$(pollutant)_per_kwh")) does not work because
+        #eval is in global scope and doesn't have access to function arguments
+        if typeof(emissions_factor_series_lb_CO2_per_kwh) == Float64
+            emissions_series_dict["CO2"] = repeat([emissions_factor_series_lb_CO2_per_kwh], 8760*time_steps_per_hour)
+        elseif length(emissions_factor_series_lb_CO2_per_kwh) == 1
+            emissions_series_dict["CO2"] = repeat(emissions_factor_series_lb_CO2_per_kwh, 8760*time_steps_per_hour)
+        elseif length(emissions_factor_series_lb_CO2_per_kwh) / time_steps_per_hour ≈ 8760
+            emissions_series_dict["CO2"] = emissions_factor_series_lb_CO2_per_kwh
+        elseif isempty(emissions_factor_series_lb_CO2_per_kwh)
+            emissions_series_dict["CO2"] = emissions_series("CO2", region_abbr, time_steps_per_hour=time_steps_per_hour)
+        else
+            @error "Provided emissions_factor_series_lb_CO2_per_kwh does not match the time_steps_per_hour."
```

Comment:
do all of these `@error`s in the `ElectricUtility` constructor need to be thrown (i.e. will downstream code break)? We could instead take a default action like setting the emissions factors to zero or use the `emission_series` look-up and `@warn` instead. Not sure which would be more appropriate.

Reply:
I will update so that the AVERT emissions series is used when the provided array has an invalid length, and just give a warning.